include/boost/gil/image_processing/morphology.hpp

```diff
@@ -0,0 +1,131 @@
+//
+// Copyright 2019 Ayush Bansal <abansal1008@gmail.com>
+//
+// Use, modification and distribution are subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+#ifndef BOOST_GIL_IMAGE_PROCESSING_MORPHOLOGY_HPP
+#define BOOST_GIL_IMAGE_PROCESSING_MORPHOLOGY_HPP
+
+#include <boost/assert.hpp>
+#include <boost/gil.hpp>
```

Comment:
We should avoid including the all-in-one header `#include <boost/gil.hpp>`
If this PR is accepted, it will include `morphology.hpp` leading to cyclic dependency.
Could you try to include individual headers of stuff that is used (yes, it may be tricky)?

Reply:
@mloskot I had a little doubt about this. After I removed `#include <boost/gil.hpp>`, my example and test files compiled without any errors. I was under the impression that definitions for concepts such as `ImageViewConcept` would be included from the all-in-one header, but I guess that is not the case. There should have been an error according to me, but wasn't. Is there something that I am missing?

Reply:
@ayushbansal07 You are correct to be suspicious.

I have just fixed one issue in `Jamfile`, https://github.com/boostorg/gil/commit/d026d50e178eecf6d2468f955ccaf4983ada9663, so please update your branch to get this fix.

Now, if you build the standalone headers tests - tests that verify each header can be compiled alone - `b2 cxxstd=11 test/core/headers`, then you should see that your header does not compile:

![image](https://user-images.githubusercontent.com/80741/75064367-17b2e500-54e7-11ea-85be-049588dd666e.png)

I also tried to compile your `example/morphology.cpp` using MSVC from VS2019 and it is also failing:

![image](https://user-images.githubusercontent.com/80741/75064655-a3c50c80-54e7-11ea-8f42-768b6460b007.png)

Reply:
I have now fixed all the errors and warnings for both example and test file.

Reply:
Thanks